modules/data-streams/src/internalClusterTest/java/org/elasticsearch/datastreams/DataStreamIT.java

```diff
@@ -1778,21 +1778,9 @@ public void testRemoveGhostReference() throws Exception {
                 @Override
                 public ClusterState execute(ClusterState currentState) throws Exception {
                     DataStream original = currentState.getMetadata().dataStreams().get(dataStreamName);
-                    DataStream broken = new DataStream(
-                        original.getName(),
-                        List.of(new Index(original.getIndices().get(0).getName(), "broken"), original.getIndices().get(1)),
-                        original.getGeneration(),
-                        original.getMetadata(),
-                        original.isHidden(),
-                        original.isReplicated(),
-                        original.isSystem(),
-                        original.isAllowCustomRouting(),
-                        original.getIndexMode(),
-                        original.getLifecycle(),
-                        original.isFailureStore(),
-                        original.getFailureIndices(),
-                        null
-                    );
+                    DataStream broken = new DataStream.Builder(original).setIndices(
```

Comment:
Nit: I like having a static method for creating a new builder, calling the constructor then becomes `DataStream.builder(original)`.

But this is just a preference. In the code we use both patterns as far as I know.

Reply:
Any reasoning for why you prefer that? Out of curiosity :)

Reply:
Of course, 
- I find it nice in IDE to be able to write `DataStream.` and immediately find the builder constructor. This way, I check if there is a builder and use it in one step. Otherwise, I see there is a `DataStream.Builder` and then I have to go back and write `new`
- I think it's more concise to write `DataStream.builder(original)` than `new DataStream.Builder(original)`.

But again, I understand this is my process and my taste, so I am just offering it as an alternative.

Reply:
Ah, right, okay. Can I propose another option 😉:
```java
public DataStream.Builder builder() {
    return new DataStream.Builder(this);
}
```
I think I'd prefer that over `DataStream.builder(original)`, as the latter is basically just an alias of the builder's constructor, so then I see slightly less value. But, being able to create a builder straight from a data stream instance for copying purposes, seems like a valuable "shortcut". What do you think?

Reply:
I like the idea in general, not sure if we do this in other places, definitely worth to follow it up.

Reply:
While searching for `new Builder(this)`, I only get two results: https://github.com/elastic/elasticsearch/blob/9f3cbc120033e869a18349f79e9b6c107fbc5e22/x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/common/http/HttpRequest.java#L294-L297
https://github.com/elastic/elasticsearch/blob/9f3cbc120033e869a18349f79e9b6c107fbc5e22/server/src/main/java/org/elasticsearch/cluster/metadata/ComposableIndexTemplate.java#L358-L360
I also see things like:
https://github.com/elastic/elasticsearch/blob/9f3cbc120033e869a18349f79e9b6c107fbc5e22/server/src/main/java/org/elasticsearch/client/internal/support/AbstractClient.java#L440-L442
but that feels like a slightly different use case/scenario. Do you have any preference for the method name? Both `toBuilder` and `copy` work for me. Or is your opinion that we shouldn't do it because it's not widely done in the project?

Reply:
Yeah I've seen that pattern in several places. Does your motivation for wanting such a static method align with what Mary mentioned? Or are there other reasons you'd like to have such a method (other than that it's seen in other places)? As I said before, I see relatively little value in these methods, as they, IMO, are only marginally more concise than the constructor itself. OTOH, they're also not that verbose so it's definitely not something I'm going to block this PR over for much longer.